feat/Login Page v2

OrPerl opened the pull request:
### Please check if the PR fulfills these requirements
- [X] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

#### What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)
This PR is purely a feature PR.

**Before (left), After (right)**
<p float="left">
<img src="https://user-images.githubusercontent.com/72373284/186692607-44bb343a-1a12-4403-a402-7533e43eca86.png" height="350px" width="200px" />

<img src="https://user-images.githubusercontent.com/72373284/186692964-a6328b35-39fb-4726-8c58-07015df03a8e.png"  height="350px" width="200px" />
</p>

#### What is the current behavior? (You can also link to an open issue here)
The current application doesn't have translations, form validation is implemented manually and the design components are basic HTML with sprinkled CSS.

The header is a very basic text as a logo.
Desktop browsers were not taken into account for the design/behavior of the page.

See Issues:
- https://github.com/aburd/beats-with-friends/issues/15
- https://github.com/aburd/beats-with-friends/issues/16
- https://github.com/aburd/beats-with-friends/issues/17

#### What is the new behavior (if this is a feature change)?
- Added solid-i18next for translation support.
- Added Hope-UI for design components.
- Added felte/Solid for form validation.
- Refactored the Login page code/JSX/CSS to change both how form validations are being conducted and the look&feel.
- Added support for SVGO and a new logo in both SVG and PNG.
- Changed the page design to work on both mobile and desktop.

#### Does this PR introduce a breaking change? (What changes might users need to make in their application due to this PR?)
This PR changes the behavior of the Login page including validations for the email and password fields.
While the Login page is a standalone page, the password validation change can affect existing customers who created weaker password and will have an issue to login.

Following this PR, the signup form should change to reflect the password validation requirement for newly registered users.

#### Other information:
None.

OrPerl committed:
Login v2

- Added support for i18n using i18next.
- Added hope-ui for design.
- Added felte/solid for form validation.
- Refactored the LoginPage to use the newly added libraries and update the design (CSS).
- Added support for SVGO.
- Added a new logo in both SVG and PNG.
- Updated the logo for the Login Page.
- Updated tsConfig settings.

OrPerl committed:
Merge branch 'master' into feat/login-page-redesign

aburd commented on package.json:
```diff
@@ -10,15 +10,26 @@
+    "@felte/validator-yup": "^1.0.9",
+    "@hope-ui/solid": "^0.6.7",
     "@solidjs/router": "^0.4.2",
+    "@stitches/core": "^1.2.8",
```
Is this the theming library?

OrPerl replied:
@aburd Good catch.
I am not sure how it got it, but after confirming that Hope-UI is working without it and reinstalling hope-ui does not re-add it, I decided to remove it.

aburd commented on src/i18n/config.ts:
```diff
@@ -0,0 +1,29 @@
+import i18next from 'i18next';
```
I thought you said we didn't need a backend anymore and that it would embed the translations into the JS?

OrPerl replied:
@aburd Sorry, my explanation was probably not clear.
If you refer to https://github.com/i18next/i18next-http-backend.
What it does is run a small server along your application to consume the translation files.
This does not require any special configurations on your dev environment or production (I have used in several projects in the past).

This can also be replaced with a CDN package if you prefer to not have it running in the code.
This is just to simplify translations load, however if you do not want them to run through a server, it can be setup differently.

aburd replied:
Thanks for the explanation! I got it. I'll resolve this.

aburd commented on src/pages/Login/LoginPage.tsx:
```diff
@@ -0,0 +1,105 @@
+import beatWithFriendsLogoUrl from '@/assets/img/beatWithFriendsLogo.png';
+import useTranslation from "@/i18n/useTranslation";
+
+YupPassword(yup);
```
Do we want to explicitly call this every time? Or just call it in one place?
What happens with TS? How does it know that the objects have been extended?

OrPerl replied:
@aburd 
For this PR I only focused on the Login page.
If we start working on a signup page, then it would be better to have this run only once at the start of the application.

In terms of Typescript, the plugin extends yup implementation by exposing the password field for string (`string().password()`).

OrPerl committed:
Removed  "@stitches/core"

- Not sure at the moment how it got included but does not seem to have any connections to hope-ui (tried to remove and reinstall hope-ui, but this was not included).

aburd reviewed (approved):
You may fire when ready!
